src/nftables.c

```diff
@@ -0,0 +1,371 @@
+  const void *data;
+  uint32_t len;
+
+  if (!nftnl_rule_is_set(nlr, NFTNL_RULE_USERDATA))
+    return NULL;
+
+  data = nftnl_rule_get_data(nlr, NFTNL_RULE_USERDATA, &len);
+
+  if (nftnl_udata_parse(data, len, parse_rule_udata_cb, tb) < 0)
+    return NULL;
+
+  if (!tb[NFTNL_UDATA_RULE_COMMENT])
+    return NULL;
+
+  return strdup(nftnl_udata_get(tb[NFTNL_UDATA_RULE_COMMENT]));
+} /* nftnl_rule_get_comment */
+
+static void submit(const char *table, const char *chain, char *comment,
+                   uint64_t bts, uint64_t cnt) {
+  value_list_t vl = VALUE_LIST_INIT;
+  int status;
+
+  sstrncpy(vl.plugin, "nftables", sizeof(vl.plugin));
+
+  status = ssnprintf(vl.plugin_instance, sizeof(vl.plugin_instance), "%s-%s",
+                     table, chain);
+  if ((status < 1) || ((unsigned int)status >= sizeof(vl.plugin_instance)))
+    return;
+
+  sstrncpy(vl.type_instance, comment, sizeof(vl.type_instance));
+
+  sstrncpy(vl.type, "ipt_bytes", sizeof(vl.type));
+  vl.values = &(value_t){.derive = (derive_t)bts};
+  vl.values_len = 1;
+  plugin_dispatch_values(&vl);
+
+  sstrncpy(vl.type, "ipt_packets", sizeof(vl.type));
+  vl.values = &(value_t){.derive = (derive_t)cnt};
+  plugin_dispatch_values(&vl);
+} /* void submit */
+
+static int submit_cb(struct nftnl_expr *e, void *data) {
+  const char *name = nftnl_expr_get_str(e, NFTNL_EXPR_NAME);
+  uint64_t bytes, count;
+
+  struct nftnl_rule *r = (struct nftnl_rule *)data;
+  const char *table = nftnl_rule_get_str(r, NFTNL_RULE_TABLE);
+  const char *chain = nftnl_rule_get_str(r, NFTNL_RULE_CHAIN);
+  char *comment = nftnl_rule_get_comment(r);
+  /* const uint64_t handle = nftnl_rule_get_u64(r, NFTNL_RULE_HANDLE); */
+
+  if (strcmp(name, "counter") == 0) {
+    count = nftnl_expr_get_u64(e, NFTNL_EXPR_CTR_PACKETS);
+    bytes = nftnl_expr_get_u64(e, NFTNL_EXPR_CTR_BYTES);
+    plugin_log(LOG_NOTICE, "Table: %s | Chain: %s | Comment: %s", table, chain,
+               comment);
+    plugin_log(LOG_NOTICE, "Bytes: %lu | Packets: %lu", bytes, count);
+    submit(table, chain, comment, bytes, count);
+  }
+  free(comment);
+
+  return MNL_CB_OK;
+} /* int submit_cb */
+
+static int table_cb(const struct nlmsghdr *nlh, void *data) {
+  struct nftnl_rule *t;
+  char *comment = (char *)data;
+  char *t_comment = NULL;
+
+  t = nftnl_rule_alloc();
+  if (t == NULL) {
+    ERROR("nftables plugin: Error allocating nftnl_rule");
+    goto err;
+  }
+
+  if (nftnl_rule_nlmsg_parse(nlh, t) < 0) {
+    perror("nftnl_rule_nlmsg_parse");
+    goto err_free;
+  }
+
+  t_comment = nftnl_rule_get_comment(t);
+  plugin_log(LOG_NOTICE, "table_cb | filter_comment: %s rule_comment: %s",
+             comment, t_comment);
+  if (strlen(comment) > 0) {
+    if (t_comment && strcmp(t_comment, comment) == 0) {
+      nftnl_expr_foreach(t, submit_cb, t);
+    }
+  } else if (t_comment) {
+    nftnl_expr_foreach(t, submit_cb, t);
+  }
+
+err_free:
+  if (t_comment)
+    free(t_comment);
+  nftnl_rule_free(t);
```

Comment:
`nftnl_rule_free` here frees the allocated rule previously above `t = nftnl_rule_alloc();`

This is not consistent with other similar cases, in which I free the comment either way, even if its NULL. So for consistency I should either check every similar case or in the other case let `free` when ptr is NULL.